### PR TITLE
implement authentication

### PR DESCRIPTION
**Pull request by Jlrine2**

### CODE REVIEW

File: api/hyp3_api/auth.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+from os import environ
+
+import jwt
+
+
+def decode_token(token, required_scopes):
+    try:
+        jwt.decode(token, environ['AUTH_PUBLIC_KEY'], algorithms=environ['AUTH_ALGORITHM'])
+    except (jwt.DecodeError, jwt.ExpiredSignatureError):
+        return {
```
Comment (asjohnston-asf) [inline api/hyp3_api/auth.py:11]: formatting could be improved here, `return {'active': True}` would be fine

File: api/hyp3_api/auth.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import time
+from os import environ
+
+import jwt
+
+
+def decode_token(token, required_scopes):
+    try:
+        jwt.decode(token, environ['AUTH_PUBLIC_KEY'], algorithms=environ['AUTH_ALGORITHM'])
```
Comment (asjohnston-asf) [inline api/hyp3_api/auth.py:9]: `decode()` can deduce the algorithm for RSA256, dunno if `environ['AUTH_ALGORITHM']` should be optional or not, probably better to make it required

Comment (Jlrine2) [reply api/hyp3_api/auth.py:9]: It is going to be made required in a future version of PyJWT so is reccomended to be provided

File: api/hyp3_api/auth.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    except (jwt.DecodeError, jwt.ExpiredSignatureError):
+        return {
+            'active': False
+        }
+    return {
+        'active': True
+    }
+
+
+def get_mock_jwt_cookie(user, expires_in=100):
```
Comment (asjohnston-asf) [inline api/hyp3_api/auth.py:19]: I like units on variable names where possible, `lifetime_in_seconds` or the like

File: api/hyp3_api/openapi-spec.yml [hunk 4]
```diff
@@ -19,6 +19,9 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/job"
+      security:
+        - URSAuth: []
```
Comment (asjohnston-asf) [inline api/hyp3_api/openapi-spec.yml:23]: `earthdata_login` is a better name

File: api/hyp3_api/openapi-spec.yml [hunk 5]
```diff
@@ -46,3 +49,10 @@ components:
       required:
         - granule
       additionalProperties: false
+
+  securitySchemes:
+    URSAuth:
+      type: apiKey
+      name: asf-urs
+      in: cookie
```
Comment (asjohnston-asf) [inline api/hyp3_api/openapi-spec.yml:57]: I'd move `in` before `name` so it reads more naturally

File: api/hyp3_api/auth.py (hunk 3 above)
Comment (Jlrine2) [reply api/hyp3_api/auth.py:19]: Sounds good, I just stole that from Datapool

**Pull request merged**